### PR TITLE
refresh collation version when pulp_ansible is installed

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: definitions/procedures/restore/reindex_databases.rb [hunk 1]
```diff
@@ -16,6 +17,11 @@ def run
 
         spinner.update('Reindexing the databases')
         execute!('runuser - postgres -c "reindexdb -a"')
+        if check_min_version('python3.11-pulp-ansible', '0.20.0')
+          execute!('runuser -c '\
+                   '\'echo "ALTER COLLATION pulp_ansible_semver REFRESH VERSION;"'\
+                   '| psql pulpcore\' postgres')
```
Comment (ekohl) [inline definitions/procedures/restore/reindex_databases.rb:23]: This isn't needed for remote DBs?

**Review by ekohl — COMMENTED** [review]
Do you have a reference on why this is needed?

File: definitions/procedures/restore/reindex_databases.rb (hunk 1 above)
Comment (evgeni) [reply definitions/procedures/restore/reindex_databases.rb:23]: Not when they don't switch the OS :)

**Comment by evgeni** [comment]
> Do you have a reference on why this is needed?

pulp_ansible ships an "own" collation since https://github.com/pulp/pulp_ansible/commit/d442dded1e4831e4788d4f79007ebc7b0c3df0ea, which requires a "refresh" of the version (after the reindex) if the underlying collation library (icu) changes, which happens during 8to9. otherwise you get "WARNING:  collation "pulp_ansible_semver" has version mismatch" in the postgres logs.

we do the same in leapp.

**Review by ehelms — APPROVED** [review]

**Review by ekohl — APPROVED** [review]
I don't see why a custom collation is needed, but given it's used I think this is correct.

**Pull request merged**